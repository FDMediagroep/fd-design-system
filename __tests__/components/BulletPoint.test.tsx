import React from 'react';
import {
    BulletPoint,
    getCssClassNames,
} from '../../src/components/bullet-point/BulletPoint';
import { render } from '@testing-library/react';

describe('BulletPoint', () => {
    test('should render correctly', () => {
        const { getByText } = render(
            <BulletPoint bullets={['one', 'two', 'three']} />
        );

        expect(getByText('one')).toBeTruthy();
        expect(getByText('two')).toBeTruthy();
        expect(getByText('three')).toBeTruthy();
    });

    test('should align correctly', () => {
        const { container } = render(
            <BulletPoint alignment="block" bullets={['one', 'two', 'three']} />
        );
        const { container: container2 } = render(
            <BulletPoint alignment="right" bullets={['one', 'two', 'three']} />
        );

        expect(container).toMatchSnapshot();
        expect(container2).toMatchSnapshot();
    });

    test('should render variants correctly', () => {
        const { container } = render(
            <BulletPoint
                variant="variant-1"
                bullets={['one', 'two', 'three']}
            />
        );

        expect(container).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(
            <BulletPoint
                variant="variant-1"
                bullets={['one', 'two', 'three']}
            />
        );

        expect(
            container.firstElementChild?.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
