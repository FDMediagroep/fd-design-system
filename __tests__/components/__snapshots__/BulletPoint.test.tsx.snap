// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPoint should align correctly 1`] = `
<div>
  <div
    class="bulletPoint "
  >
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        one
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        two
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        three
      </span>
    </p>
  </div>
</div>
`;

exports[`BulletPoint should align correctly 2`] = `
<div>
  <div
    class="bulletPoint right"
  >
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        one
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        two
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        three
      </span>
    </p>
  </div>
</div>
`;

exports[`BulletPoint should render variants correctly 1`] = `
<div>
  <div
    class="bulletPoint "
    data-variant="variant-1"
  >
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        one
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        two
      </span>
    </p>
    <p>
      <span
        class="icon"
      >
        <test-file-stub />
      </span>
      <span>
        three
      </span>
    </p>
  </div>
</div>
`;
